### PR TITLE
Update field data behavior

### PR DESCRIPTION
**Pull request by calvn**

### CODE REVIEW

File: logical/framework/field_data.go [hunk 1]
```diff
@@ -243,24 +247,33 @@ func (d *FieldData) getPrimitive(k string, schema *FieldSchema) (interface{}, bo
...
 		if err != nil {
-			return nil, true, err
+			return nil, false, err
 		}
 		if err := decoder.Decode(raw); err != nil {
-			return nil, true, err
+			return nil, false, err
+		}
+		if len(result) == 0 {
+			return make([]int, 0), true, nil
```
Comment (kalafut) [inline logical/framework/field_data.go]: Why is this the only type of slice type that is converting nil slices into empty slices?

Comment (calvn) [reply logical/framework/field_data.go]: Other non-string slice types are also doing this conversion: https://github.com/hashicorp/vault/pull/5489/files#diff-e0df0d262e580544c11a06ad128a7aeeR265, https://github.com/hashicorp/vault/pull/5489/files#diff-e0df0d262e580544c11a06ad128a7aeeR275.

String slice types are doing this as well, but indirectly (through `strutil.TrimStrings` with calls `make([]string, len(items))`)

Comment (calvn) [reply logical/framework/field_data.go]: Fixed

**Comment by kalafut** [comment]
LGTM

**Pull request merged**